### ISSUE TITLE
Unable to use custom resolvers

### ISSUE THREAD

**Issue by rohitgirdhar** [omry/omegaconf#576]
**Describe the bug**
Tried following documentation [here](https://omegaconf.readthedocs.io/en/latest/usage.html#custom-interpolations), it returns a string instead of the final value. I was trying to do this hydra (like https://github.com/facebookresearch/hydra/issues/861), but realize I am not able to do it in normal OmegaConf either.

**To Reproduce**
```python
$ python
Python 3.8.5 (default, Sep  4 2020, 07:30:14)
[GCC 7.3.0] :: Anaconda, Inc. on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> from omegaconf import OmegaConf
>>> OmegaConf.register_new_resolver("plus", lambda x, y: x + y)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
AttributeError: type object 'OmegaConf' has no attribute 'register_new_resolver'
>>> OmegaConf.register_resolver("plus", lambda x, y: x + y)
>>> c = OmegaConf.create({"a": 1,"b": 2,"a_plus_b": "${plus:${a},${b}}"})
>>> c.a_plus_b
'${plus:1,2}'
>>> print(int(c.a_plus_b)) 
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
ValueError: invalid literal for int() with base 10: '${plus:1,2}'
>>> import omegaconf; omegaconf.__version__
'2.1.0dev19'
>>>
```

**Expected behavior**
It should return the number 3 as the final resolved value 

**Additional context**
- [x] OmegaConf version: 2.1.0dev19
- [x] hydra version: 1.1.0dev3
- [x] Python version: 3.8.5
- [x] Operating system : Ubuntu
- [x] Please provide a minimal repro

**Comment by rohitgirdhar** [comment]
Just uninstalled and re-installed hydra and it seems to work now..
```bash
$ pip uninstall hydra-core
$ pip install hydra-core --upgrade --pre
$ python
>>> import hydra; hydra.__version__
1.1.0dev4
>>> import omegaconf; omegaconf.__version__
2.1.0dev20
```

**Comment by omry** [comment]
OmegaConf nested interpolations are available as of 2.1.0dev20.
